sentry-ruby/lib/sentry/transport.rb

```diff
@@ -33,6 +49,23 @@ def send_event(event)
       event
     end
 
+    def is_rate_limited?(event_type)
+      # check category-specific limit
+      delay =
+        case event_type
+        when "event"
+          # confusing mapping, but it's decided by Sentry
+          @rate_limits["error"]
+        when "transaction"
+          @rate_limits["transaction"]
+        end
+
+      # check universal limit if not category limit
+      delay ||= @rate_limits[nil]
+
+      !!delay && delay > Time.now
```

Comment:
Please check both the universal and the category-based limit. The stored category limit could be lower than the universal limit.